File: sample/use_cocoapods/objectivec/GNAdSampleBanner/GNAdSampleBanner/ViewController.m
//
//  ViewController.m
//  GNAdSampleBanner
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    _adView = [[GNAdView alloc] initWithAdSizeType:GNAdSizeTypeTall appID:@"YOUR_ZONE_ID"];
    _adView.delegate = self;
    _adView.rootViewController = self;
    //_adView.GNAdlogPriority = GNLogPriorityInfo;

    [self.view addSubview:_adView];
    _adView.center = CGPointMake(self.view.center.x, self.view.center.y);
    [_adView startAdLoop];
}

- (void)dealloc
{
    [_adView startAdLoop];
    _adView.delegate = nil;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (BOOL)shouldStartExternalBrowserWithClick:(GNAdView *)adView landingURL:(NSString *)landingURL
{
    NSLog(@"ViewController: shouldStartExternalBrowserWithClick: %@.",landingURL);
    return YES;
}

@end

File: sample/swift/GNAdSampleBanner/GNAdSampleBanner/ViewController.swift
//
//  ViewController.swift
//  GNAdSampleBanner
//

import UIKit

class ViewController: UIViewController, GNAdViewDelegate {
    
    var _adView : GNAdView = GNAdView(adSizeType:GNAdSizeTypeSmall, appID:"YOUR_ZONE_ID")

    override func viewDidLoad() {
        super.viewDidLoad()
        
        _adView.delegate = self
        _adView.rootViewController = self
        //_adView.bgColor = YES
        //_adView.geoLocationEnable = YES
        //_adView.gnAdlogPriority = GNLogPriorityInfo
        _adView.center = CGPoint(x: self.view.center.x, y: self.view.center.y);
        
        self.view.addSubview(_adView)
        _adView.startAdLoop()
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    func shouldStartExternalBrowserWithClick(nativeAd: GNAdView, landingURL: String) -> Bool {
        NSLog("ViewController: shouldStartExternalBrowserWithClick : %@.", landingURL);
        return true;
    }

}
